### PR TITLE
fuzz: add test for the gossipd message processing

### PR DESCRIPTION
**Pull request by NishantBansal2003**
Add state machine fuzz tests for gossipd message processing. Currently, all message states handled by gossipd are covered here, but other gossip messages like `gossip_timestamp_filter`, `query_short_channel_id`, and `query_channel_range` are handled by connectd, so those states need to be explicitly tested in separate fuzz tests.

Most of the work here is based on https://github.com/ElementsProject/lightning/pull/8423, so I've added the PR author as a co-author. I also rebased the changes and fixed the entropy issue afterward. Additionally, some new states have been added (e.g. UTXO lookup, update blockheight, seeker state machine), and a few existing states have been updated to support e2e message processing (e.g. UTXO lookup in channel announcements).

Also, in the ref PR there was a discussion about issues with multi worker fuzzing. I’ve addressed that in https://github.com/ElementsProject/lightning/commit/26f7b522eb88897854180f27e8fe27b625fffaf6. I think with this we can run regression tests on some previously observed edge-cases

## Checklist
Before submitting the PR, ensure the following tasks are completed. If an item is not applicable to your PR, please mark it as checked:

- [x] The changelog has been updated in the relevant commit(s) according to the [guidelines](https://docs.corelightning.org/docs/coding-style-guidelines#changelog-entries-in-commit-messages).
- [x] Tests have been added or modified to reflect the changes.
- [x] Documentation has been reviewed and updated as needed.
- [x] Related issues have been listed and linked, including any that this PR closes.
- [x] *Important* All PRs must consider how to reverse any persistent changes for `tools/lightning-downgrade`

### CODE REVIEW

File: gossipd/gossip_store.c [hunk 1]
```diff
@@ -19,7 +19,8 @@
 /* Obsolete ZOMBIE bit */
 #define GOSSIP_STORE_ZOMBIE_BIT_V13 0x1000U
 
-#define GOSSIP_STORE_TEMP_FILENAME "gossip_store.tmp"
+#define GOSSIP_STORE_TEMP_FILENAME    tal_strcat(tmpctx, GOSSIP_STORE_FILENAME, ".tmp")
+#define GOSSIP_STORE_CORRUPT_FILENAME tal_strcat(tmpctx, GOSSIP_STORE_FILENAME, ".corrupt")
```
Comment (ddustin) [inline gossipd/gossip_store.c]: The correct C way to do this is:
```c
#define GOSSIP_STORE_TEMP_FILENAME    GOSSIP_STORE_FILENAME ".tmp"
```

Comment (NishantBansal2003) [reply gossipd/gossip_store.c]: Yes, but that would fail in our fuzzing setup, where we need to define a different `GOSSIP_STORE_FILENAME` for each worker to avoid corruption from multiple fuzz workers accessing the same file. String concatenation only works with string literals, but here we need `GOSSIP_STORE_FILENAME` to be a variable so that it can vary across workers

Comment (ddustin) [reply gossipd/gossip_store.c]: In general caps macros (particularly without parens) imply something constant. Replacing it with something dynamic, particularly one dependent on tmpctx can easily throw someone off in the future causing hard-to-discover bugs and issues down the line.

What's the issue that's causing multiple fuzz workers to access the same file?

Comment (NishantBansal2003) [reply gossipd/gossip_store.c]: > What's the issue that's causing multiple fuzz workers to access the same file?

In libFuzzer, multiple fuzz workers run in the same CWD. If there is a file with the same name, one worker may rename it to .tmp, .compact, or something while another worker is still accessing the original file. In that case, the second worker may panic when attempting to access the file, causing a false-positive fuzz failure to be reported





> In general caps macros (particularly without parens) imply something constant. Replacing it with something dynamic, particularly one dependent on tmpctx can easily throw someone off in the future causing hard-to-discover bugs and issues down the line.

Got your concern. I've updated the approach so the production paths stay as literal constants (no behaviour change), and are only overridden when `GOSSIP_STORE_FUZZ_OVERRIDE` is defined
